test/aircraft/FlightManagementSystem/Fms.spec.js

```diff
@@ -819,6 +820,74 @@ ava('._prependLegCollectionWithRouteAmendment() adds LegModels for each routeStr
     t.true(fms.legCollection[1].routeString === routeAmmendmentMock[1]);
 });
 
+ava('._regenerateSidLeg() creates a new SID leg of identical route string for the currently assigned departure runway', (t) => {
+    const initialRunwayMock = airportModelFixture.getRunway('19L');
+    const nextRunwayMock = airportModelFixture.getRunway('19R');
+    const fms = buildFmsMockForDeparture();
+    const replaceDepartureProcedureSpy = sinon.spy(fms, 'replaceDepartureProcedure');
+    const sidLegIndex = fms._findLegIndexForProcedureType(PROCEDURE_TYPE.SID);
+    const oldWaypoints = fms.legCollection[sidLegIndex].waypointCollection;
+
+    t.true(_isEqual(fms.departureRunwayModel, initialRunwayMock));
+    t.true(oldWaypoints[0].name === 'fixix');
+    t.true(oldWaypoints[1].name === 'roppr');
+    t.true(oldWaypoints[1].altitudeMaximum === 7000);
+    t.true(oldWaypoints.length === 7);
+
+    fms.departureRunwayModel = nextRunwayMock;
+    fms._regenerateSidLeg();
+
+    const newWaypoints = fms.legCollection[sidLegIndex].waypointCollection;
+
+    t.true(newWaypoints[0].name === 'jaker');
+    t.true(newWaypoints[1].name === 'roppr');
+    t.true(newWaypoints[1].altitudeMaximum === 7000);
```

Comment:
if we have a waypoint `roppr` couldn't we also assume that we have the altitude restriction as well?  The thing we are testing here is that the flight plan has been changed, thats it.  Restrictions and such would likely be tested elsewhere.

I don't mind the verbosity here, but I think this test would be just as effective with:
```javascript
t.true(newWaypoints[0].name === 'jaker');
t.true(newWaypoints.length === 7);
t.true(replaceDepartureProcedureSpy.calledWithExactly(routeString, departureModel);
```